src/commands/generate/models.ts

```diff
@@ -116,8 +113,8 @@ export default class Models extends Command {
     switch (language) {
     case Languages.typescript:
       fileGenerator = new TypeScriptFileGenerator({
-        modelType: tsModelType as 'class' | 'interface',
-        enumType: tsEnumType as 'enum' | 'union',
+        modelType: tsModelType as undefined | 'class' | 'interface',
+        enumType: tsEnumType as undefined | 'enum' | 'union'
```

Comment:
Dont cast, you have default values.

Reply:
just removed undefined from line 117 then it gave me 2 check fail and when I removed it from line 116 then it gave 3 check fails

Reply:
Thats because you have not added all the default values back as I suggested in https://github.com/asyncapi/cli/pull/480#discussion_r1142244719

There are about 5 or something left 🙂

Reply:
ok fixing it